### DISCUSSION TITLE
Verification of SA-NEG-QCR2000-R with Joukowsky airfoil

### DISCUSSION THREAD

**Discussion by rois1995** [su2code/SU2#2376] [General]
Hi everyone,

I have tried to verify the implementation of the SA turbulence model when the Neg, QCR2000 and Rotation corrections (CRot=1) are applied, following the work in https://doi.org/10.2514/6.2023-1244 . Only portion of these results will be shown in the SciTech paper regarding the HLPW5 results. The fully-structured meshes are provided by the HiFi-CFD Verification Workshop and taken from https://github.com/HighFidelityCFDVerificationWorkshop/2024RANS/tree/main/Joukowski_Mesh (you have to generate those using python). All of the rate of convergence plots are done on the absolute error with respect to a reference value (which is not theoretical but is the result obtained from the SANS code on an adapted mesh using P3 polynomials, which you can find in the above reference). 

Reference results:
1. CD = 0.007872;
2. CDp = 0.001244;
3. CDv = 0.006627;

The simulations are considered converged if the log of the density RMS residual is lower than -13.

The flow conditions are:
1. Mach number Ma = 0.15
2. Re number = 6 million
3. T = 520 °R (288.889 K)
4. AoA = 0 deg

The RANS equations are solved in non-dimensional formulation with the freestream_vel_eq_mach option.

First I tried running with Roe+MUSCL and different limiters (also not activating the limiter) and with the central scheme of JST. The limiters I've used are the Venkatakrishnan (V), Venkatakrishnan-Wang (VW), Nishikawa R3 (R3).  Green Gauss has been used for the gradient computations.

Convergence of the drag coefficient
![Conv_RoeVWVsRoeR3VsRoeNoLimVsJST](https://github.com/user-attachments/assets/5b7b3a9c-caee-4651-8a7a-5a731411adc0)

Convergence of the residuals for the density and for nu-tilde
![ConvRMSnutildeAndRho_RoeVWVsRoeR3VsRoeNoLimVsJST](https://github.com/user-attachments/assets/e9c1a1bc-755d-4c05-a21a-46419b82e713)

Convergence of the pressure component of the drag coefficient
![ConvCDp_RoeVWVsRoeR3VsRoeVVsRoeNoLimVsJST](https://github.com/user-attachments/assets/49608071-9ba4-4939-b82b-08df21d376e2)

Convergence of the viscous component of the drag coefficient
![ConvCDv_RoeVWVsRoeR3VsRoeVVsRoeNoLimVsJST](https://github.com/user-attachments/assets/996db48d-0de4-4a13-b190-18e6f8ba58e6)

First of all, no solution was obtained with the Roe+MUSCL and no limiter (which I will call No Lim. from now on) on the first two meshes due to divergence problems. Nevertheless, the best results wrt the drag coefficient on the finest meshes are the ones without limiter and with JST. Convergence is of second order for JST and No Lim. on the finest meshes (No Lim. gets worse on the last mesh and goes back to first order). When considering the pressure and viscous components of the drag coefficient, however, we can see that the ones computed by SU2 converge to different values with respect to the ones computed by Fun3D and SANS. This makes the error go with 0-order on the finest meshes for JST and No Lim., whereas all the others retain first order convergence (the same seems to happen to Fun3D for the Cdv).

I then tried focusing on the VW limiter and changing the value of the Venka coefficient (From now on I will report only the Cd and residual. If other plots are needed don't hesitate to ask for them).

Convergence of the drag coefficient
![Conv_VWCoeff](https://github.com/user-attachments/assets/9c8149a5-1f3a-46bd-9537-80d33966ca7f)

Convergence of the residuals
![ConvRMSnutildeAndRho_VWCoeff](https://github.com/user-attachments/assets/73f35cd6-2484-48b5-ac4a-b3fa20bc9613)

As soon as the coefficient is increased with respect to the 0.03 value the results improve a lot, both in terms of residual convergence (on the finest meshes) and in terms of actual values. The convergence to a "wrong" value of Cdp and Cdv still remains.

Finally (not really, but I don't want to make it too long), I tried converting the structured mesh into a full triangular one by diagonalizing the quads in Pointwise. No results have been obtained with the GG method on these grids due to divergence problems. Thus I tried with the weighted least squares (WLS) and compared against the structured results. Three limiters are considered: Venka-Wang (VW), Nishikawa R3 (R3) and Nishikawa R5 (R5). Convergence problems on the triangulated meshes start from L2 mesh (for VW, for R3 and R5 from L4 mesh), where the residuals stagnate. Moreover, the results on the two finest meshes are completely wrong.

Convergence of drag coefficient
![Conv_StructVsTria](https://github.com/user-attachments/assets/bc89ee2c-07ca-4f21-a791-ee826a6ff1f7)

Convergence of residuals
![ConvRMSnutildeAndRho_StructVsTria](https://github.com/user-attachments/assets/27c135e4-1dae-4176-84b7-d4a15f30dd2f)

All of the simulations have been performed on the same number of cores, and with the same version of SU2 (latest develop with CRot = 1).

In conclusion:
1. SU2 reaches 2nd order convergence only with Roe+MUSLC and no limiter or JST.
2. Using limiters worsen the convergence even if the Ma is low and the AoA is 0 (is it supposed to happen or wrong implementation of limiters?).
3. "Wrong" values of Cdp and Cdv are reached (compared to Sans P3 results).
4. Using triangulated grids destroys convergence on fine meshes and completely misses the solution.

At the following link you can check all of these results. Plus, structured and triangulated grids are provided as well as the series of configs used.
https://polimi365-my.sharepoint.com/:f:/g/personal/10507725_polimi_it/EvaNl7DMLy9Eu-_JH3i0xvcBflGUa37eIfnuU-_29MjpDA?e=4O2cHJ

**Comment by pcarruscag** [comment]
To get a totally systematic refinement when using Vekatakrishan-type limiters you need to reduce the coefficient as the mesh size is reduced. Otherwise, for the same coefficient the limiters are less active on a fine mesh than on a coarse one, because the solution range in the vicinity of a node decreases while the reference value remains the same.
This behavior is good from the point of view of not impacting the order of accuracy (or even making it seem higher) but it may explain why your finer meshes are having those issues.

**Reply by rois1995** [reply]
I also recalled something like that and indeed I also tried changing the coefficients as the mesh is refined. In particular, I tried both increasing (VarCoeff) and decreasing (VarCoeff Dim) and I must say that the best results are obtained with an increase of the coefficient. First order is obtained on the finest mesh by the VarCoeff, but it just means that the error is kind of saturated. Nevertheless, the improvement with increasing the coefficient as the meshes are refined is clear across the board: better convergence of the residuals and convergence to the "correct" value.

The coefficients in increasing order are  0.1, 0.2, 0.4, 0.8, 1.6, 3.2 (in reverse order for the other case).

Convergence of drag coefficient
![Conv_VWCoeff_WithLim](https://github.com/user-attachments/assets/a3fd439f-ea52-4740-a23a-38dd68ea8dcf)

Convergence of the residuals
![ConvRMSnutildeAndRho_VWCoeff_WithLim](https://github.com/user-attachments/assets/ef10620d-c4bc-4f78-99f8-3d0dde1dc10b)
